bigtable/tests/unit/test_row_data.py

```diff
@@ -813,11 +813,102 @@ def test_build_updated_request(self):
         expected_result = _ReadRowsRequestPB(table_name=self.table_name,
                                              filter=row_filter.to_pb(),
                                              rows_limit=6)
-        expected_result.rows.row_ranges.add(**self.row_range1.
-                                            get_range_kwargs())
+        expected_result.rows.row_ranges.add(
+            start_key_open=last_scanned_key,
+            end_key_open=self.row_range1.end_key
+        )
+
+        self.assertEqual(expected_result, result)
+
+    def test_build_updated_request_no_start_key(self):
+        from google.cloud.bigtable.row_filters import RowSampleFilter
+        row_filter = RowSampleFilter(0.33)
+        last_scanned_key = b"row_key25"
+        request = _ReadRowsRequestPB(filter=row_filter.to_pb(),
+                                     rows_limit=8,
+                                     table_name=self.table_name)
+        request.rows.row_ranges.add(end_key_open=b"row_key29")
+
+        request_manager = self._make_one(request, last_scanned_key, 2)
+
+        result = request_manager.build_updated_request()
+
+        expected_result = _ReadRowsRequestPB(table_name=self.table_name,
+                                             filter=row_filter.to_pb(),
+                                             rows_limit=6)
+        expected_result.rows.row_ranges.add(start_key_open=last_scanned_key,
+                                            end_key_open=b"row_key29")
+
+        self.assertEqual(expected_result, result)
+
+    def test_build_updated_request_no_end_key(self):
+        from google.cloud.bigtable.row_filters import RowSampleFilter
+        row_filter = RowSampleFilter(0.33)
+        last_scanned_key = b"row_key25"
+        request = _ReadRowsRequestPB(filter=row_filter.to_pb(),
+                                     rows_limit=8,
+                                     table_name=self.table_name)
+        request.rows.row_ranges.add(start_key_closed=b"row_key20")
+
+        request_manager = self._make_one(request, last_scanned_key, 2)
+
+        result = request_manager.build_updated_request()
+
+        expected_result = _ReadRowsRequestPB(table_name=self.table_name,
+                                             filter=row_filter.to_pb(),
+                                             rows_limit=6)
+        expected_result.rows.row_ranges.add(start_key_open=last_scanned_key)
```

Comment:
Do we have a testcase which reproduces the "whole table read" issue from #5876?  I.e., with no start or end key specified.

Reply:
@tseaver, yes.  `test_build_updated_request_plain` is that test.  

@AVaksman, can you please move `test_build_updated_request_plain` up to line 823, and call it `test_build_updated_request_full_table`?

Reply:
Yes, "whole table read" is reproduced in testcase [`test_build_updated_request_plain`](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/5966/files/a43c4c0d328618fd4c174a4e922504594c0953ee#diff-e1d4ea33ffcf7ba98a70d7dc9af8ec6eR900)

Also there is "rows limit only" test (no row_keys or row_ranges) [`test_build_updated_request_rows_limit`](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/5966/files/a43c4c0d328618fd4c174a4e922504594c0953ee#diff-e1d4ea33ffcf7ba98a70d7dc9af8ec6eR887)